testsuite/openshift/metrics/prometheus.py

```diff
@@ -70,24 +72,38 @@ def get_metrics(self, key: str = "", labels: dict[str, str] = None) -> Metrics:
 
         return Metrics(response.json()["data"]["result"])
 
-    def wait_for_scrape(self, target_service: str, metrics_path: str = "/metrics"):
+    @backoff.on_predicate(backoff.constant, interval=10, jitter=None, max_tries=35)
+    def is_reconciled(self, service_monitor: ServiceMonitor):
+        """True, if all endpoints in ServiceMonitor are active targets"""
+        scrape_pools = set(target["scrapePool"] for target in self.get_active_targets())
+        endpoints = len(service_monitor.model.spec.endpoints)
+        for i in range(endpoints):
+            if f"serviceMonitor/{service_monitor.namespace()}/{service_monitor.name()}/{i}" not in scrape_pools:
+                return False
+
+        return True
+
+    def wait_for_scrape(self, service_monitor: ServiceMonitor, metrics_path: str):
         """Wait before next metrics scrape on service is finished"""
         call_time = datetime.now(timezone.utc)
 
-        @backoff.on_predicate(backoff.constant, interval=10, jitter=None, max_tries=20)
+        @backoff.on_predicate(backoff.constant, interval=10, jitter=None, max_tries=4)
         def _wait_for_scrape():
             """Wait for new scrape after the function call time"""
-            for target in self.get_targets():
+            for target in self.get_active_targets():
                 if (
-                    "service" in target["labels"].keys()
-                    and target["labels"]["service"] == target_service
-                    and target["discoveredLabels"]["__metrics_path__"] == metrics_path
+                    f"serviceMonitor/{service_monitor.namespace()}/{service_monitor.name()}" in target["scrapePool"]
+                    and metrics_path in target["scrapeUrl"]
                 ):
                     return call_time < datetime.fromisoformat(target["lastScrape"][:26]).replace(tzinfo=timezone.utc)
             return False
 
-        assert _wait_for_scrape(), "Prometheus didn't reconcile ServiceMonitor in time"
+        assert _wait_for_scrape(), "Scrape wasn't done in time"
+
+    def commit(self):
+        self._client = KuadrantClient(headers=self.headers, verify=False)
+        self.client = ApyProxy(self.url, self._client).api.v1
 
-    def close(self):
-        """Close httpx client connection"""
-        self._client.close()
+    def delete(self):
+        if self._client is not None:
+            self._client.close()
```

Comment:
I don't like much the use of commit/delete functions inside the client-like objects. In my vision, they should commit some data to the server and remove data from it. I'd create a separate interface for client-like objects (KuadrantClient also might be under it) and give it open(?) and close methods. Or maybe httpx already has something like this.  (Interface Segregation or whatever ;) )

Reply:
We already have close() so making it a `LifecycleObject` brings it in line with our other stuff. I can revert this change for now and we can discuss later